sdk/eventhub/azure-eventhub/samples/async_samples/recv_with_checkpoint_by_batch_async.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+An example to show receiving events from an Event Hub with checkpoint store doing checkpoint by batch asynchronously.
+In the `receive` method of `EventHubConsumerClient`:
+If no partition id is specified, the checkpoint_store are used for load-balance and checkpoint.
+If partition id is specified, the checkpoint_store can only be used for checkpoint.
+"""
+
+import asyncio
+import os
+from collections import defaultdict
+from azure.eventhub.aio import EventHubConsumerClient
+from azure.eventhub.extensions.checkpointstoreblobaio import BlobCheckpointStore
+
+CONNECTION_STR = os.environ["EVENT_HUB_CONN_STR"]
+STORAGE_CONNECTION_STR = os.environ["AZURE_STORAGE_CONN_STR"]
+BLOB_NAME = "your-blob-name"  # Please make sure the blob resource exists.
+
+partition_recv_cnt_since_last_checkpoint = defaultdict(int)
+checkpoint_batch_event_cnt = 20
+
+
+async def on_event(partition_context, event):
+    # put your code here
+    p_id = partition_context.partition_id
+    print("Received event from partition: {}".format(p_id))
+    partition_recv_cnt_since_last_checkpoint[p_id] += 1
+    if partition_recv_cnt_since_last_checkpoint[p_id] >= checkpoint_batch_event_cnt:
+        await partition_context.update_checkpoint(event)
+        partition_recv_cnt_since_last_checkpoint[p_id] -= checkpoint_batch_event_cnt
+
+
+async def receive(client):
+    """
+    Without specifying partition_id, the receive will try to receive events from all partitions and if provided with
+    a checkpoint store, the client will load-balance partition assignment with other EventHubConsumerClient instances
+    which also try to receive events from all partitions and use the same storage resource.
+    """
+    await client.receive(on_event=on_event)
+    # With specified partition_id, load-balance will be disabled
```

Comment:
Just for clarity, I would add "for example" to the end:
```
# With specified partition_id, load-balance will be disabled, for example:
# await client.receive(on_event=on_event, partition_id='0'))
```

Reply:
updated :)